common/hyperscaler/account_pool.go

```diff
@@ -99,25 +131,33 @@ func (p *secretBindingsAccountPool) IsSecretBindingUsed(hyperscalerType Type, te
 		return false, nil
 	}
 
-	shootlist, err := p.gardenerClient.Resource(gardener.ShootResource).Namespace(p.gardenerNS).List(context.Background(), metav1.ListOptions{})
-	if err != nil {
-		return false, fmt.Errorf("listing Gardener shoots: %w", err)
-	}
+	return p.bindingsClient.IsUsedByShoot(secretBinding)
+}
 
-	for _, shoot := range shootlist.Items {
-		sh := gardener.Shoot{Unstructured: shoot}
-		if sh.GetSpecSecretBindingName() == secretBinding.GetName() {
-			return true, nil
-		}
+func (sp *secretBindingsAccountPool) SharedCredentialsSecretBinding(hyperscalerType Type, euAccess bool) (*gardener.SecretBinding, error) {
+	// selector
+	shared := true
+	selector := getLabelsSelector(hyperscalerType, shared, euAccess)
+
+	// get binding
+	secretBindings, err := sp.bindingsClient.GetSecretBindings(selector)
+	if err != nil {
+		return nil, fmt.Errorf("getting secret binding: %w", err)
 	}
 
-	return false, nil
+	return sp.bindingsClient.GetLeastUsed(secretBindings)
 }
 
-func (p *secretBindingsAccountPool) CredentialsSecretBinding(hyperscalerType Type, tenantName string, euAccess bool) (*gardener.SecretBinding, error) {
-	labelSelector := fmt.Sprintf("tenantName=%s, hyperscalerType=%s, !dirty", tenantName, hyperscalerType.GetKey())
-	labelSelector = addEuAccessSelector(labelSelector, euAccess)
-	secretBinding, err := p.getSecretBinding(labelSelector)
+func (p *secretBindingsAccountPool) CredentialsSecretBinding(hyperscalerType Type, euAccess bool, shared bool, tenantName string) (*gardener.SecretBinding, error) {
+	// selector
+	selector := getLabelsSelector(hyperscalerType, shared, euAccess)
+
+	// label selector modifications
+	labelSelector := fmt.Sprintf("%s, tenantName=%s, !dirty", selector, tenantName)
+	labelSelector = strings.ReplaceAll(labelSelector, ", shared!=true", "")
+
+	// get binding
+	secretBinding, err := p.bindingsClient.GetSecretBinding(labelSelector)
```

Comment:
if you remove shared!=true expression then there is a risk that a secretbinding with "shared=true" label could be taken, or I'm wrong?

Reply:
In my opinion you got the point.
We need to use specifically `shared=true` or `shared!=true`. True?